src/collective/ifttt/actions/ifttt.py

```diff
@@ -0,0 +1,146 @@
+from plone.app.contentrules.actions import ActionEditForm
+from plone.app.contentrules.browser.formhelper import ContentRuleFormWrapper
+from plone.contentrules.rule.interfaces import IExecutable
+from plone.contentrules.rule.interfaces import IRuleElementData
+from zope import schema
+from zope.component import adapter
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.schema.vocabulary import SimpleTerm
+from zope.schema.vocabulary import SimpleVocabulary
+
+import logging
+
+
+logger = logging.getLogger('collective.ifttt')
+
+payload_options = SimpleVocabulary([
+    SimpleTerm(value=u'Description', title=_(u'Description of action')),
+    SimpleTerm(value=u'Username', title=_(u'Username of Editor')),
+    SimpleTerm(
+        value=u'Event start date/time', title=_(u'Event Start Date/Time')
+    ),
+])
+
+
+class IIftttTriggerAction(Interface):
+    """
+        Definition of the configuration available for a  Ifttt action
+    """
+    ifttt_name = schema.TextLine(
+        title=_(u'IFTTT event name'),
+        description=_(u'The IFTTT event name'),
+        required=False,
+    )
+
+    payload_option = schema.Choice(
+        title=_(u'Choose 2nd Payload'),
+        vocabulary=payload_options,
+        description=_(
+            u'Choose whether the 3rd payload is Description, '
+            u'Username of editor, or Event Start Date/time'
+        ),
+    )
+    payload = schema.Text(
+        title=_(u'Payload'),
+        required=False,
+        description=_(u'Provide input for chosen payload')
+    )
+
+
+@implementer(IIftttTriggerAction, IRuleElementData)
+class IftttTriggerAction(SimpleItem):
+    """
+        The implementation of the action defined before
+    """
+
+    ifttt_name = u''
+    payload_option = u''
+    payload = u''
+
+    element = 'plone.actions.Ifttt'
+
+    @property
+    def summary(self):
+        return _(
+            u'${ifttt_name} ${payload_option}',
+            mapping=dict(
+                ifttt_name=self.ifttt_name,
+                payload_option=self.payload_option,
+                payload=self.payload,
+            ),
+        )
+
+
+EXECUTOR = ThreadPoolExecutor(max_workers=1)
+
+
+@implementer(IExecutable)
+@adapter(Interface, IIftttTriggerAction, Interface)
+class IftttTriggerActionExecutor(object):
+    """
+        The executor for this action
+    """
+    timeout = 120
+
+    def __init__(self, element):
+        self.element = element
+
+    def __call__(self, *args, **kwargs):
+        r = self.element.request
+        ifttt_name = self.element.ifttt_name
+        payload_option = self.element.payload_option
+        payload = self.element.payload
+        try:
+            EXECUTOR.submit(
+                # Post HTTP request
+                r.post,
+                ifttt_name,
+                payload_option,
+                data=payload,
+                timeout=self.timeout
+            )
```

Comment:
@Shriyanshagro Please, remove that executor and try to call requests.post directly to be able to fix that. Now that translates to

```
requests.post(ifttt_name, payload_option, data=payload, timeout=self.timeout)
```

which is not a valid requests post call. It is missing IFTTT url and IFTTT secret and IFTTT payload is not read from the context to the actual value.

http://docs.python-requests.org/en/master/

Reply:
Also see Sally's comment about what should the POST be actually calling https://github.com/collective/collective.ifttt/pull/34#issuecomment-393980242

Reply:
@datakurre About POST request's URL,
what will be the role of URL filled by the user? which is our 2nd input field as per discussion in #24 
As after looking sally's https://github.com/collective/collective.ifttt/pull/34#issuecomment-393980242, there is no input from the user, other than `event name`(1st input field) and pre-stored `secret key`

Reply:
@Shriyanshagro Site Administrator may configure SECRET KEY and content rule with EVENT NAME and 3RD PAYLOAD. Then Site Administrator may assing the rule to some folder on the site. When the content rule fires, it should read SECRET KEY from registry, EVENT NAME from the rule element, TITLE and ABSOLUTE URL and selected 3RD PAYLOAD from its context object (which caused it to fire). And the POST that info to IFTTT. (I don’t know enough about IFTTT to know the exact url.)

So title and url are not filled by the user, but read from the context that triggered the content rule. I assume the url is useful at IFTTT side so the post van trigger something useful there.

PS. I might have written somewhere that description is the second payload. If I have, that is wrong., title, url and selected 3rd payload is the right set as written in stories / issues.

Reply:
Yes, I followed IFTTT documentation and finally understand it's payload and request call(url). 
Currently, I'm struggling to get `secret key`, `ifttt name`, `event start date/time` and `username`. 

I'll relook at them following your comments.